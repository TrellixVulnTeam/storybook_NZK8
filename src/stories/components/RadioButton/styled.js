// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const ButtonLabelWrapper = styled.div`
  display: flex;
  padding-top: ${({ label }) => (label ? '8px' : null)};
`;

export const RadioButtonWrapper = styled.div`
  position: relative;
  width: 16px;
  height: 16px;
  border-radius: 30px;
  border: 1px solid ${p => (p.checked ? colors.transparent : colors.silver)};
  background-color: ${({ checked }) => (checked ? colors.brightBlue : colors.white)};
  transition: all 100ms ease;
  cursor: pointer;

  /* checkmark */
  span {
    display: ${p => (p.checked ? 'block' : 'none')};
    position: absolute;
    width: 14px;
    height: 14px;
    border: 2px solid ${colors.white};
    border-radius: 50%;
    transition: all 100ms ease;
  }
`;

export const Label = styled.div`
  display: inline-block;
  margin-left: 5px;
  font-size: 14px;
  font-weight: 600;
`;
