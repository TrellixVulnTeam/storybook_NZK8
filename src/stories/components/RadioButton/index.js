// @flow

import React from 'react';
import { RadioButtonWrapper, Label, ButtonLabelWrapper } from './styled';

type Props = {
  onClick?: Function,
  label?: string,
  checked?: *,
};

export default (props: Props) => (
  <>
    <ButtonLabelWrapper label={props.label}>
      <RadioButtonWrapper onClick={props.onClick} checked={props.checked}>
        <span />
      </RadioButtonWrapper>
      {props.label && <Label>{props.label}</Label>}
    </ButtonLabelWrapper>
  </>
);
