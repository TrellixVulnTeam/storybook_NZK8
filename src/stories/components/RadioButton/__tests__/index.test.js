import React from 'react';
import { mount } from 'enzyme';
import RadioButton from 'components/RadioButton/index';

describe('<RadioButton />', () => {
  it('should render correctly without props', () => {
    const sut = mount(<RadioButton />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with full props', () => {
    const props = {
      onClick: jest.fn(),
      checked: true,
      label: 'label',
    };
    const sut = mount(<RadioButton {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
