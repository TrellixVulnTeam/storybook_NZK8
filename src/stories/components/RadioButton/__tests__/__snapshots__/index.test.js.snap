// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioButton /> should render correctly with full props 1`] = `
<_default
  checked={true}
  label="label"
  onClick={[MockFunction]}
>
  <styled__ButtonLabelWrapper
    label="label"
  >
    <div
      className="styled__ButtonLabelWrapper-sc-1u6knhx-0 jnycro"
      label="label"
    >
      <styled__RadioButtonWrapper
        checked={true}
        onClick={[MockFunction]}
      >
        <div
          checked={true}
          className="styled__RadioButtonWrapper-sc-1u6knhx-1 fmfpKP"
          onClick={[MockFunction]}
        >
          <span />
        </div>
      </styled__RadioButtonWrapper>
      <styled__Label>
        <div
          className="styled__Label-sc-1u6knhx-2 kUfDyV"
        >
          label
        </div>
      </styled__Label>
    </div>
  </styled__ButtonLabelWrapper>
</_default>
`;

exports[`<RadioButton /> should render correctly without props 1`] = `
<_default>
  <styled__ButtonLabelWrapper>
    <div
      className="styled__ButtonLabelWrapper-sc-1u6knhx-0 keXxD"
    >
      <styled__RadioButtonWrapper>
        <div
          className="styled__RadioButtonWrapper-sc-1u6knhx-1 dqisNB"
        >
          <span />
        </div>
      </styled__RadioButtonWrapper>
    </div>
  </styled__ButtonLabelWrapper>
</_default>
`;
